src/main/java/fi/aalto/cs/apluscourses/utils/async/ScheduledTaskExecutor.java

```diff
@@ -0,0 +1,23 @@
+package fi.aalto.cs.apluscourses.utils.async;
+
+import com.intellij.util.concurrency.AppExecutorUtil;
+import java.util.concurrent.TimeUnit;
+
+public class ScheduledTaskExecutor {
+
+  /**
+   * A simple wrapper on the IntelliJ IDEA's Executors class. Does nothing but scheduling a task to
+   * be run at regular intervals.
+   *
+   * @param runnable a {@link Runnable} task to execute.
+   * @param startAfter an interval to start the task after.
+   * @param repeatEvery an interval to repeat the task every.
+   * @param timeUnit a time unit to apply to previous 2 (two) parameters.
+   */
+  public ScheduledTaskExecutor(Runnable runnable, long startAfter, long repeatEvery,
+      TimeUnit timeUnit) {
+    AppExecutorUtil.getAppScheduledExecutorService()
+        .scheduleWithFixedDelay(runnable, startAfter, repeatEvery, timeUnit);
+    //todo: incremental interval how-to?
```

Comment:
the schedules overall work in the similar manner, so in order to have an increasing execution interval, one will have to go a bit deeper to the `Thread` level, and atm not sure if I want to :stuck_out_tongue: